# ハンバーガー統計学 with Python 第2章

第２回講義（2017年9月29日講義、10月3日課題提出締切）

このページは、<a href="http://kogolab.chillout.jp/elearn/hamburger/index.html" target="_blank">ハンバーガー統計学にようこそ！</a>の内容をプログラミング言語 Python に合わせて書き直したものです（著作権者の許諾を得て利用しております）。

<h1 STYLE="background: #c2edff;padding: 0.5em;"><a href="http://kogolab.chillout.jp/elearn/hamburger/chap2/sec0.html" target="_blank">第2章 信頼区間</a></h1>

<img src="image/w-potato00001.jpg">

あなたは、平均と分散という考え方を知ることによって、ポテトの長さの分布がどのようなものであるかと推測できるようになりました。

さて、前の章でデータを取るために使ったポテトはたまたまあなたが買ったひとつのポテトです。もし違う日に買いに行けば違うポテトが手に入り、したがって違うデータになります。

私たちがデータを取るのは、「手元にあるこのポテトについて知りたいのだ」という場合ももちろんありますが、多くの場合は「ワクワクバーガーで売っているポテト」や「モグモグバーガーで売っているポテト」といったものの「全体」について知りたいのです。

しかし、そのお店で毎日売られているポテトを全部調べることは実際上不可能です。それではどうしたらいいのでしょうか……。

<h2 STYLE="background: #c2edff;padding: 0.5em;"><a href="http://kogolab.chillout.jp/elearn/hamburger/chap2/sec1.html" target="_blank">2.1　平均的ポテトを推定する</a></h2>

<img src="image/w-potato00001.jpg">

あなたの買ったポテトには、全部で49本はいっていました。あなたは49本のポテトの長さの平均と分散、標準偏差を計算することができます。

さて、今度はポテトの本数が気になってきました。もしかするとあなたが買ったポテトは、普通に売られているポテトより本数が少ないかもしれません。他の人が買ったポテトには60本くらいはいっていることもあるのかもしれません。

はたして、49本という数は、この店で売られているポテトの平均的本数なのでしょうか？　店員さんに聞いてみます。

<img src="image/salesman00002.jpg">

「あの～、このポテトって、本数を数えているわけじゃないですよね。」

「もちろんです。そんなことは実際上できませんから。」

「確かに、そうですよね。」

「何か、気になることがありましたか？」

「いや、たいしたことじゃないんだけど、このポテトは全部で49本入っていたんですよ。それが、標準のものよりも多いのかな、それとも少ないのかな、と思ったもので。」

「お客さん、そのポテトの本数をわざわざ数えたんですか？　それは、すごい。」

「そう。49本はいっていたんだけど、この数字はどうなんでしょうね。」

「うーん。どうなんだろう。自分の店で作っておきながら、そういうことは気にしたことがなかったなあ。」

「聞いたのは、そんなわけなんです。」

「うーん。本当はどうなんだろうなあ……」

<h4 style="border-bottom: solid 1px black;">母集団と標本、無作為抽出</h4>

たとえば、このお店で作られるすべてのポテトについて知りたいとします。このとき、「このお店で作られるすべてのポテト」のことを「母集団」と呼びます。

しかし、たいていの場合は、母集団は数が多すぎるので、全部を調べるわけにはいきません。そこで母集団の中から限られた数のデータを取ってきます。こうして取ってくることを「抽出」（あるいは「サンプリング」）と呼びます。またこうして取ってきたデータを「標本」と呼びます。

<img src="image/fig2001.jpg">

標本を抽出するときには、それを「無作為」に（あるいはランダムに）行うことが重要です。たとえば、このお店の開店時間から10個連続で標本を抽出したとすると、それはかなり偏ったものになることが予想されます。無作為に抽出するためには、たとえば1時間ごとにひとつの標本を取り出したり、50個おきにひとつの標本を取り出したりします。サイコロや乱数表などを使うこともあります。無作為に行われた抽出を「無作為抽出」（あるいは「ランダムサンプリング」）と呼びます。

標本の中のデータの数を、「標本の大きさ」あるいは「サンプルサイズ」と呼びます。

さて、あなたはワクワクバーガーでポテトを買った人を1時間ごとに一人選び、ポテトの本数を数えさせてもらいました。こうして得られたのが次のデータです。

https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/wakuwaku_potato2.txt

このデータから、平均、分散、標準偏差を計算してみましょう。

<h4 style="padding: 0.25em 0.5em;color: #494949;background: transparent;border-left: solid 5px #7db4e6;">課題2.1</h4>

[ワクワクポテトの本数](https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/wakuwaku_potato2.txt)のデータから、ポテトの本数の平均、分散、標準偏差を求めてください。

In [1]:
def average(list): # 平均
    sum = 0
    for x in list:
        sum += x
    return sum / len(list)

In [2]:
def variance(list): # 分散
    ave = average(list)
    sum = 0
    for x in list:
        sum += (x - ave) ** 2
    return sum / len(list)

In [3]:
import math
def standard_deviation(list): # 標準偏差
    return math.sqrt(variance(list))

In [4]:
# URL によるリソースへのアクセスを提供するライブラリをインポートする。
#import urllib # Python 2 の場合
import urllib.request # Python 3 の場合

In [5]:
# 課題2.1
url = 'https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/wakuwaku_potato2.txt'
urllib.request.urlretrieve(url, 'wakuwaku_potato2.txt') 

# ダウンロードしたデータから、列ごとに数字を読み込んでリストに格納する。
col1 = [] # ０列目の数字を格納する予定のリスト
for i, line in enumerate(open('wakuwaku_potato2.txt')): # ファイルを開いて一行一行読み込む
    c = line.split() # 行を空白文字で分割したものをcというリストに入れる
    col1.append(int(c[0])) # ０列目の単語を整数に変換してcol1に入れる

print(average(col1))
print(variance(col1))
print(standard_deviation(col1))

49.2
3.1600000000000006
1.7776388834631178


<h2 STYLE="background: #c2edff;padding: 0.5em;"><a href="http://kogolab.chillout.jp/elearn/hamburger/chap2/sec2.html" target="_blank">2.2　母集団の平均と分散を推定する</a></h2>

さて、標本の平均は49.2本になりました。問題は、この値を母集団の平均とみなしていいかどうかです。たくさんある母集団の中から、たった10個だけ取り出してきた標本の平均を、母集団の平均とみなしていいのでしょうか。

これを考えるには、母集団の平均・分散と標本の平均・分散との間にどのような関係があるのかを知っておく必要があります。

<h4 style="border-bottom: solid 1px black;">標本から母集団の平均と分散を推定する</h4>

今、ポテトの母集団を考えて、そこから10個のデータを標本を抽出することを何回も繰り返したとしましょう。

10個のデータを無作為に抽出し、その平均と分散を計算します。それを標本平均1、標本分散1としておきます。次に、また別の10個のデータを無作為に抽出して、平均と分散を計算します。それを標本平均2、標本分散2としておきます。このようなことを何回も繰り返すわけです。

そうして得られた標本平均を平均すると、母集団の平均（母平均と呼びます）に等しくなります。そうなると、標本分散全体の平均も、母集団の分散（母分散と呼びます）に等しくなりそうな気がします。しかし、これは母分散よりも少し小さな値になります。

そこで、母分散の推定値として次のようなものを考えます。

<img src="image/fig2002.jpg">

　　不偏分散＝（（データ－平均値）の二乗）の総和÷（個数-1）

これは以前に出てきた分散の式で、「個数」の部分を「個数-1」にしたものです。これを「不偏分散」と呼びます。この不偏分散を「母分散の推定値」として使います。 まとめると次のようになります。

母集団全体の平均（母平均）と分散（母分散）を知りたいときに、母集団全部のデータは手に入らないことがほとんどです。そこで、母集団から無作為抽出をして標本のデータを手に入れます。そうすると標本の平均値は母平均の推定値として使えます。また、標本の不偏分散は母分散の推定値として使えます。

<center>◇</center>

それでは前に取った10個の標本データをもとにして、母集団の平均と分散を推定してみましょう。
https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/wakuwaku_potato2.txt

<h4 style="padding: 0.25em 0.5em;color: #494949;background: transparent;border-left: solid 5px #7db4e6;">課題2.2</h4>

[ワクワクポテトの本数](https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/wakuwaku_potato2.txt)のデータから、ポテトの本数の不偏分散を求めてください。

In [6]:
# 不偏分散を求める関数
def unbiased_variance(list):
    ave = average(list)
    sum = 0.0
    for x in list:
        sum += (x - ave) ** 2.0
    return sum / (len(list) - 1)

In [7]:
# 課題2.2
unbiased_variance(col1)

3.5111111111111115

私の計算では、平均値は49.2、不偏分散は3.51となりました。

　　不偏分散＝31.6÷(10-1)＝3.511111…

この結果から、母平均の推定値は49.2、母分散の推定値は3.51ということになります。あなたが買ったポテトの本数は49本だったので、平均的なポテトだということができるでしょう。

さて、標本から、ポテトの本数の推定値は49.2本であることを計算したあなたは店員さんに報告しました。

<img src="image/salesman00002.jpg">

「店員さん、10個のデータからこのお店のポテトの本数の平均を推定すると、49.2本になりました。私が買ったポテトの本数は49本でしたので、これはまあまあ平均的なポテトだということができますね。」

「へえええ。全部のポテトの本数を調べなくても、その平均値がわかるんだ。すごいねえ、統計学ってのは。」

「平均値だけでなくて分散も推定できるんですよ。」

「なんだい？　分散ってのは？」

「ええと、その、つまり、データのばらつきのことです。」

「ふうん。しかし、全体のポテトのから見れば、たった10個のポテトのデータなのに、そこから全体の平均値がわかってしまうなんて、不思議だねえ。さっき、49.2本って言ったけど、それが全体のポテトの平均値にぴったり一致するわけじゃなんだろう？」

「『ぴったり一致』か。そういわれればそうですね。どうなんだろう？」

<h2 STYLE="background: #c2edff;padding: 0.5em;"><a href="http://kogolab.chillout.jp/elearn/hamburger/chap2/sec3.html" target="_blank">2.3　区間推定／信頼区間</a></h2>

母集団から標本を取り出して計算した標本平均は、母平均の推定値として使うことができます。しかし、それは母平均にぴったり一致しているわけではありません。あくまでも推定値です。サンプルサイズが小さければ、標本平均と母平均が離れている可能性が高くなりそうですし、逆に、サンプルサイズが大きければ、標本平均と母平均とが近くなるような気がします。

そこで、母平均を、ある幅を持って推定しようということを考えます。これを「区間推定」と呼びます。「標本から推定すると、母平均はこの値からこの値までの間にはいるのではないか」という形で推定をおこなうのです。

<h4 style="border-bottom: solid 1px black;">区間推定の考え方</h4>

区間推定の考え方を説明していきましょう。 まずポテトの母集団の分布を考えます。この分布が「正規分布」にしたがっているとします。正規分布というのは、下の図のように平均を山の中心として左右になめらかに広がった「つりがね型」をした分布です。

<img src="image/fig2003.jpg">

次に、サンプルサイズを10として、母集団から10個のデータを取ります。この平均を計算して、それを標本平均1とします。標本を戻して、もう一度10個のデータを取ります。その平均を標本平均2とします。このようなことを何回も繰り返します。

このとき標本平均の分布を描いてみると、これもまた正規分布に従います。そのときの平均値は母平均に一致します。また分散は母分散の10分の1、つまり「(サンプルサイズ)分の1」になります。

<img src="image/commonfig00001.jpg">

いま、標本平均をひとつ取ったとすると、その値は下の標本平均の分布図の矢印の範囲内に大体はいります。もちろん平均に近い値になることが多く、それから外れた裾野の方の値になることはめったにありません。

<img src="image/fig2005.jpg">

ここまで、標本を何回も取ることを考えてきましたが、通常私たちが標本を取るのは一回限りのことです。そこで、一回限りの標本を取ったときに、その平均値が推定したい母平均を含んでいるような範囲はどうなるかということを考えます。たとえば、標本平均が下の図のように母平均から少しずれていたとしたら矢印の範囲を指定すれば、母平均を含んでいることになります。

<img src="image/fig2006.jpg">

次の問題は、母平均を含むような範囲を決めるときに、どのような範囲にすればいいかということです。統計学では、伝統的に「95%の確率で母平均が含まれるような範囲」を使います。もっと厳しくしたいときには「99%の確率で母平均が含まれるような範囲」を使います。これをそれぞれ「95%信頼区間」、「99%信頼区間」と呼びます。

<img src="image/fig2007.jpg">

母平均から標本平均を左右にずらしていきます。左右の標本平均で囲まれた部分の面積が全体の95%になるまでずらしていきます。そのとき、その標本平均から母平均までの差を取り、それを標本平均の範囲とすれば、その範囲に母平均が含まれる確率がちょうど95%になります。この原理で、95%信頼区間を求めます。

<h4 style="border-bottom: solid 1px black;">信頼区間の求め方</h4>

信頼区間を計算していきましょう。 まず、母平均の推定値として、標本平均を使います。 次に、標本平均の分布の分散の推定値として、母分散を標本数で割ったものを使います。ここで、母分散は不偏分散で推定します。つまり、次のようになります。

　　標本平均の分散＝（母分散／サンプルサイズ）＝（不偏分散／標本数）

　　標本平均の標準偏差＝（不偏分散／サンプルサイズ）の平方根

　　標本平均の標準偏差を「標本標準誤差」（SE=standard error）と呼びます。

さて、これで信頼区間を求めることができます。

　　信頼区間＝標本平均±t×標本標準誤差

このとき「t」は信頼区間で決められた分布の面積が95%になるような数値です。この「t」の具体的な値を決めるために、t分布を変形して、平均が0、標準偏差が1のt分布を作ります。t分布は正規分布に似ていますが、サンプルサイズによって形が少し変わります（後述します）。

<img src="image/fig2008.jpg">

サンプルサイズ10のときのt分布を作ってみると、-tからtまでで挟まれた面積が全体の95%になるようなtの値は2.262になります。また、99%の場合は3.250になります。

これで信頼区間の計算ができます。標本平均は49.2でしたので

　　信頼区間＝49.2±2.262×標本標準誤差

不偏分散は3.51でしたので、

標本標準誤差＝sqrt(3.51／10)　　★sqrtは平方根を求める関数です★

　　　　　　＝0.592

よって

信頼区間＝49.2±2.262×0.592

　　　　＝49.2±1.34

　　　　＝47.86～50.54

つまり、47.86から50.54が95%信頼区間になります。この意味は、母平均が95%の確率で47.86から50.54までの間に含まれるということです。

<center>◇</center>

それでは今度は99%信頼区間を計算してみましょう。99%のときのtの値は3.250になりますので、

信頼区間＝49.2±3.250×標本標準誤差

不偏分散は3.51でしたので、

標本標準誤差＝sqrt(3.51／10)＝0.592

よって

信頼区間＝49.2±3.250×0.592

　　　　＝49.2±1.924

　　　　＝51.124～47.276

<h4 style="border-bottom: solid 1px black;">t分布表と自由度</h4>

さて、信頼区間を計算したあなたは再び店員さんに報告しに行きました。

<img src="image/salesman00002.jpg">

「10個の標本データからこのお店全体のポテトの本数の平均を推定すると、49.2本になりました。もちろんこれは推定値なので、ぴったりこの値になるとは限りません。そこで95%の確率で母平均を含んでいるような範囲を計算します。それは、47.86本から50.54本の間だということがわかりました。」

「へえええ。こりゃまたすごいねえ。母平均の範囲までわかるんだ。でも、「95%の確率で」っていうのは一体誰が決めたんだい？」

「ん～。95%の確率で母平均がこの範囲にあるということは、5%の確率で、この範囲の外にある場合もあるということです。しかし、それは5%の確率でしか起こらないわけですからめったには起こりません。そこで、95%の確率でよしとしようということです。」

「いいかえれば、20回に1回はハズレになるわけですが、それでよしとするということです。もちろんもっと厳しくしたいという要望があれば、99%を基準に取ることもできます。その場合は100回に1回しかハズレになりません。ですからより堅実に母平均のはいりそうな範囲を推定することができます。」

t分布は正規分布に似たつりがね型の分布ですが、サンプルサイズによって少しずつ形が違っています。標本数が小さいときは、正規分布よりも平べったい形になります。逆にサンプルサイズが大きいときは正規分布に近づいていきます。サンプルサイズが無限大のとき、正規分布に一致します。

<img src="image/fig2009.jpg">

信頼区間を計算するためには、サンプルサイズに応じてt分布の95%と99%の値を知る必要があります。そこで下のようなt分布表がよく使われます。

<center>t分布表</center>

<table width="50%" border="1" align="center" cellpadding="2" cellspacing="0">
  <tr bgcolor="#FFFFCC"> 
    <td width="32%"> 
      <div align="right">自由度</div></td>
    <td width="34%"> 
      <div align="right">確率95%</div></td>
    <td width="34%"> 
      <div align="right">確率99%</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">1</div></td>
    <td><div align="right">12.706</div></td>
    <td><div align="right">63.657</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">2</div></td>
    <td><div align="right">4.303</div></td>
    <td><div align="right">9.925</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">3</div></td>
    <td><div align="right">3.182</div></td>
    <td><div align="right">5.841</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">4</div></td>
    <td><div align="right">2.776</div></td>
    <td><div align="right">4.604</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">5</div></td>
    <td><div align="right">2.571</div></td>
    <td><div align="right">4.032</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">6</div></td>
    <td><div align="right">2.447</div></td>
    <td><div align="right">3.707</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">7</div></td>
    <td><div align="right">2.365</div></td>
    <td><div align="right">3.499</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">8</div></td>
    <td><div align="right">2.306</div></td>
    <td><div align="right">3.355</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">9</div></td>
    <td><div align="right">2.262</div></td>
    <td><div align="right">3.250</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">10</div></td>
    <td><div align="right">2.226</div></td>
    <td><div align="right">3.169</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">11</div></td>
    <td><div align="right">2.201</div></td>
    <td><div align="right">3.106</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">12</div></td>
    <td><div align="right">2.179</div></td>
    <td><div align="right">3.055</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">13</div></td>
    <td><div align="right">2.160</div></td>
    <td><div align="right">3.021</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">14</div></td>
    <td><div align="right">2.145</div></td>
    <td><div align="right">2.977</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">15</div></td>
    <td><div align="right">2.131</div></td>
    <td><div align="right">2.947</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">16</div></td>
    <td><div align="right">2.120</div></td>
    <td><div align="right">2.921</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">17</div></td>
    <td><div align="right">2.110</div></td>
    <td><div align="right">2.898</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">18</div></td>
    <td><div align="right">2.101</div></td>
    <td><div align="right">2.878</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">19</div></td>
    <td><div align="right">2.093</div></td>
    <td><div align="right">2.861</div></td>
  </tr>
  <tr> 
    <td bgcolor="#eeeeee"> 
      <div align="right">20</div></td>
    <td><div align="right">2.086</div></td>
    <td><div align="right">2.845</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">21</div></td>
    <td><div align="right">2.080</div></td>
    <td><div align="right">2.831</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">22</div></td>
    <td><div align="right">2.074</div></td>
    <td><div align="right">2.819</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">23</div></td>
    <td><div align="right">2.069</div></td>
    <td><div align="right">2.807</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">24</div></td>
    <td><div align="right">2.064</div></td>
    <td><div align="right">2.797</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">25</div></td>
    <td><div align="right">2.060</div></td>
    <td><div align="right">2.787</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">26</div></td>
    <td><div align="right">2.056</div></td>
    <td><div align="right">2.779</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">27</div></td>
    <td><div align="right">2.052</div></td>
    <td><div align="right">2.771</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">28</div></td>
    <td><div align="right">2.048</div></td>
    <td><div align="right">2.763</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">29</div></td>
    <td><div align="right">2.045</div></td>
    <td><div align="right">2.756</div></td>
  </tr>
  <tr> 
    <td bordercolor="#CCCCCC" bgcolor="#eeeeee"> 
      <div align="right">30</div></td>
    <td><div align="right">2.042</div></td>
    <td><div align="right">2.750</div></td>
  </tr>
  <tr> 
    <td bordercolor="#CCCCCC" bgcolor="#eeeeee"> 
      <div align="right">40</div></td>
    <td><div align="right">2.021</div></td>
    <td><div align="right">2.704</div></td>
  </tr>
  <tr> 
    <td bordercolor="#CCCCCC" bgcolor="#eeeeee"> 
      <div align="right">60</div></td>
    <td><div align="right">2.000</div></td>
    <td><div align="right">2.660</div></td>
  </tr>
  <tr> 
    <td bordercolor="#CCCCCC" bgcolor="#eeeeee"> 
      <div align="right">120</div></td>
    <td><div align="right">1.980</div></td>
    <td><div align="right">2.617</div></td>
  </tr>
  <tr> 
    <td bordercolor="#CCCCCC" bgcolor="#eeeeee"> 
      <div align="right">∞</div></td>
    <td><div align="right">1.960</div></td>
    <td><div align="right">2.576</div></td>
  </tr>
</table>

この表で「自由度」というのはサンプルサイズから1を引いたものです。たとえばサンプルサイズが10のとき、自由度は9になります。

なお、この表に自由度が見あたらないときは、近い自由度の値を使ってください。たとえば、自由度32であれば、近い自由度である30を使います。実用上はそれで問題はありません。

また、300や500というような大きな自由度の時は「∞（無限大）」の値を使ってください。

<center>◆</center>

それでは、ここまでの内容をまとめてみましょう。

<h4 style="border-bottom: solid 1px black;">まとめ――母集団、標本、無作為抽出、信頼区間</h4>

* それについて知りたいと思う、全体のデータを母集団という

* 母集団からいくつか取り出したデータを標本という

* そのデータの数をサンプルサイズという

* よい標本を取り出すためには、無作為（ランダム）に抽出することが必要

* 母平均は標本の平均で推定できる

* 母分散は標本の不偏分散で推定できる

* 不偏分散＝（（データ－平均値）の二乗）の総和／（個数-1）

* 母平均が95%の確率で含まれているような範囲を95%信頼区間という

* 信頼区間＝標本平均±t×標本標準誤差

* 標本標準誤差＝（不偏分散／サンプルサイズ）の平方根

* tの値は、確率（95%や99%）と自由度によって変わるので、t分布表を見る

* サンプルサイズから1引いたものを自由度という

<h2 STYLE="background: #c2edff;padding: 0.5em;"><a href="http://kogolab.chillout.jp/elearn/hamburger/chap2/sec4.html" target="_blank">2.4　実践編：平均的チキンを推定する</a></h2>

さて、ポテトの本数の信頼区間を計算することができたあなたは、店員さんからさらに頼まれました。

<img src="image/salesman00002.jpg">

「ひとつお願いがあるんだけど。」

「なんでしょう？」

「いや、実はね、この店で売っている6個入りのチキンのことなんだけどね。最近、「分量が少ない」という苦情がはいったんだよ。チキンの形はひとつひとつ違うから、もちろん一定の重さにはならないんだけど、全体として本社基準の重さからあまりずれていると問題になるんだ。ここはひとつ信頼区間というやつで、検証できないかな？」

「できると思いますよ。そのためにはチキンの重さのデータを無作為抽出で取ることが必要です。１時間おきにひとつのチキンを標本データとして取って、その重さをはかっておいてください。」

<img src="image/w-chicken.jpg">

さて、翌日、次のようなデータが手に入りました。6個入りチキンの重さのデータです。サンプルサイズは8です。

https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/fried_chicken2.txt

このデータを使って、信頼区間を計算していきましょう。

<h4 style="padding: 0.25em 0.5em;color: #494949;background: transparent;border-left: solid 5px #7db4e6;">課題2.4</h4>

[ワクワクチキンのグラム数](https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/fried_chicken2.txt)のデータを用いて、以下の問いに答えてください。ただし、t分布表は下表のとおりとします。
* 標本平均、不偏分散、標本標準誤差、確率95%のt値、確率99%のt値、95%信頼区間、99%信頼区間を求めてください。
* 標本の平均と不偏分散と標本標準誤差から何がわかるか説明してください。
* 95%信頼区間と99%信頼区間の値から何がわかるか説明してください。

<table width="50%" border="1" align="center" cellpadding="2" cellspacing="0">
  <tr bgcolor="#FFFFCC"> 
    <td width="32%"> <div align="right">自由度</div></td>
    <td width="34%"> <div align="right">確率95%</div></td>
    <td width="34%"> <div align="right">確率99%</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">1</div></td>
    <td><div align="right">12.706</div></td>
    <td><div align="right">63.657</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">2</div></td>
    <td><div align="right">4.303</div></td>
    <td><div align="right">9.925</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">3</div></td>
    <td><div align="right">3.182</div></td>
    <td><div align="right">5.841</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">4</div></td>
    <td><div align="right">2.776</div></td>
    <td><div align="right">4.604</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">5</div></td>
    <td><div align="right">2.571</div></td>
    <td><div align="right">4.032</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">6</div></td>
    <td><div align="right">2.447</div></td>
    <td><div align="right">3.707</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">7</div></td>
    <td>
<div align="right">2.365</div></td>
    <td>
<div align="right">3.499</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">8</div></td>
    <td><div align="right">2.306</div></td>
    <td><div align="right">3.355</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">9</div></td>
    <td><div align="right">2.262</div></td>
    <td><div align="right">3.250</div></td>
  </tr>
  <tr> 
    <td bgcolor="#CCCCCC"><div align="right">10</div></td>
    <td><div align="right">2.226</div></td>
    <td><div align="right">3.169</div></td>
  </tr>
</table>

In [8]:
# 課題2.4
url = 'https://raw.githubusercontent.com/maskot1977/ipython_notebook/master/toydata/fried_chicken2.txt'
urllib.request.urlretrieve(url, 'fried_chicken2.txt')

col1 = [] # ０列目の数字を格納する予定のリスト
for i, line in enumerate(open('fried_chicken2.txt')): # ファイルを開いて一行一行読み込む
    c = line.split() # 行を空白文字で分割したものをcというリストに入れる
    col1.append(int(c[0])) # ０列目の単語を整数に変換してcol1に入れる

import math
def sample_standard_error(list):
    return math.sqrt(unbiased_variance(col1)/len(list))

# ワクワクチキンの重さの
print (average(col1)) # 標本平均
print (unbiased_variance(col1)) # 不偏分散
print (sample_standard_error(col1)) # 標本標準誤差

550.375
445.69642857142856
7.464050748181484


In [9]:
# 自由度（サンプルサイズ - 1）は 7 なので、上の表から
# 確率95%のt値 = 2.365
# 確率99%のt値 = 3.499

In [10]:
# 95%信頼区間
print (average(col1) - 2.365 * sample_standard_error(col1), average(col1) + 2.365 * sample_standard_error(col1))

532.7225199805508 568.0274800194492


In [11]:
# 95%信頼区間
print (average(col1) - 3.499 * sample_standard_error(col1), average(col1) + 3.499 * sample_standard_error(col1))

524.258286432113 576.491713567887


<h4 style="border-bottom: solid 1px black;">解答例</h4>

さて、あなたの計算結果はどうなりましたか。私の計算では、次のようになりました（小数点以下1,2位を四捨五入）。

* 平均＝ 550.4

* 不偏分散＝ 445.7

* 標本標準誤差＝ 7.46

自由度は(8-1)で7です。その時の

* 確率95%のt＝ 2.365

* 確率99%のt＝ 3.499

* 95%信頼区間＝550.4±2.365×7.46 ＝532.7～568.0

* 99%信頼区間＝550.4±3.499×7.46 ＝524.3～576.5

これらの値から何がわかるかを次のように書いてみました。

・標本の平均と不偏分散と標本標準誤差から何がわかるか

<blockquote>標本のチキンの重さの平均は、550.4であった。これは母平均の推定値として使うことができる。また、不偏分散は、445.7であった。これは母分散の推定値として使うことができる。<br><br>

標本標準誤差は7.46であった。これは、母集団から8個のデータを何度も取り出して、その平均を計算したとき、その平均値の分布の標準偏差の推定値になる。</blockquote>

・95%信頼区間と99%信頼区間の値から何がわかるか？

<blockquote>95%信頼区間は532.7～568.0であった。これは、この範囲内に母平均が含まれる確率が95%であることを示している。<br><br>

99%信頼区間は524.3～576.5であった。これは、この範囲内に母平均が含まれる確率が99%であることを示している。</blockquote>

<h2 STYLE="background: #c2edff;padding: 0.5em;">ハンバーガー統計学 第２章 課題</h2>

__課題2.1__、__課題2.2__、および __課題2.4__ を解いて、指定のメールアドレスまでメールしてください。メール送信後は、エラーが帰ってきてないことを確認してください（メールアドレスを間違える人がときどき居ます）。
* 締切：10月3日（今すぐでなくても結構です）
* メールタイトル：「ハンバーガー統計学 第２章」
* 学籍番号と氏名を明記すること。
* 感想などがあれば書いてくれると嬉しいです。次回以降の講義の改善につながるかも知れません。

In [12]:
# ハンバーガー統計学 第２章 課題

<h2 STYLE="background: #c2edff;padding: 0.5em;">終わったら、<a href="HamburgerStatistics03.ipynb">第３章：カイ２乗検定</a> に進んでください。</h2>
（2017年10月3日講義、10月6日課題提出締切）